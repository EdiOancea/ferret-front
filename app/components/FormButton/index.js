import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = theme => ({
  submit: {
    marginTop: theme.spacing.unit * 3,
  },
});

const FormButton = ({ text, classes }) => (
  <Button
    type="submit"
    variant="contained"
    color="primary"
    className={classes.submit}
    fullWidth
  >
    {text}
  </Button>
);

FormButton.propTypes = {
  text: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(FormButton);
