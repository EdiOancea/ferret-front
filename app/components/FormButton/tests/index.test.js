import React from 'react';
import { render } from 'enzyme';
import FormButton from 'components/FormButton';

describe('<FormButton />', () => {
  it('should not regress', () => {
    const tree = render(<FormButton text="test text" />);
    expect(tree).toMatchSnapshot();
  });
});
