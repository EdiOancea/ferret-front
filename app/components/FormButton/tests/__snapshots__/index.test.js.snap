// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormButton /> should not regress 1`] = `
<button
  class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-contained-13 MuiButton-containedPrimary-14 MuiButton-raised-16 MuiButton-raisedPrimary-17 MuiButton-fullWidth-27 FormButton-submit-1"
  tabindex="0"
  type="submit"
>
  <span
    class="MuiButton-label-3"
  >
    test text
  </span>
</button>
`;
